mm2src/mm2_bitcoin/script/src/script.rs

```diff
@@ -420,6 +420,7 @@ impl Script {
                     _ => unreachable!(), // because we are relying on script_type() checks here
                 })
                 .map(|public| {
+                    // DISCUSS: Should we really reduce this to pkh? what is it used for?
                     vec![ScriptAddress::new_p2pkh(AddressHashEnum::AddressHash(
                         public.address_hash(),
                     ))]
```

Comment:
It's used in `addresses_from_script` which is used for getting the `from` and `to` fields of transactions details to save in history db/storage. Since we don't display p2pk addresses as pubkeys but as p2pkh addresses, it's fine to not include the pubkeys in the addresses that are used for this purpose. So it's ok to reduce this to pkh and leave it as is, we have to watch out and not use it in any other context though, which we will probably never do.